packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Expressions/SystemExtensibleSnippets.SystemModelSnippets.cs

```diff
@@ -12,6 +14,111 @@ internal partial class SystemExtensibleSnippets
+                  // TO-DO: Call the base model ctor for now until the model properties are serialized https://github.com/microsoft/typespec/issues/3330
+                  Snippets.Return(new NewInstanceExpression(returnType, new ValueExpression[] { new StringLiteralExpression(iModelInterface.Name, false) })),
+                  CSharpMethodKinds.IModelSerialization
+                );
+            }
+
+            /// <summary>
+            /// Builds the deserialization method for the I model interface.
+            /// </summary>
+            public override CSharpMethod? BuildIModelDeserializationMethod(CSharpType iModelInterface)
+            {
+                var typeOfT = TryGetModelArgumentType(iModelInterface);
+                // IPersistableModel<T>.Create(BinaryData data, ModelReaderWriterOptions options)
+                return new CSharpMethod
+                (
+                  new MethodSignature(nameof(IPersistableModel<object>.Create), null, null, MethodSignatureModifiers.None, typeOfT, null, new[] { _dataParameter, _serializationOptionsParameter }, ExplicitInterface: iModelInterface),
+                  // TO-DO: Call the base model ctor for now until the model properties are serialized https://github.com/microsoft/typespec/issues/3330
+                  Snippets.Return(new NewInstanceExpression(typeOfT, Array.Empty<ValueExpression>())),
+                  CSharpMethodKinds.IModelDeserialization
+                );
+            }
+
+            /// <summary>
+            /// Builds the get format method for the I model interface.
+            /// </summary>
+            public override CSharpMethod? BuildIModelGetFormatMethod(CSharpType iModelInterface, ValueExpression wireFormat)
+            {
+                var typeOfT = TryGetModelArgumentType(iModelInterface);
+                // ModelReaderWriterFormat IPersistableModel<T>.GetFormatFromOptions(ModelReaderWriterOptions options)
+                return new CSharpMethod
+                (
+                  new MethodSignature(nameof(IPersistableModel<object>.GetFormatFromOptions), null, null, MethodSignatureModifiers.None, typeof(string), null, new[] { _serializationOptionsParameter }, ExplicitInterface: iModelInterface),
+                  wireFormat,
+                  CSharpMethodKinds.IModelGetFormat
+                );
+            }
+
+            /// <summary>
+            /// Attempts to get the model argument type from the model interface.
+            /// </summary>
+            /// <param name="modelInterface">The <see cref="CSharpType"/> that represents the model interface.</param>
+            /// <returns>The first argument type of <paramref name="modelInterface"/>.</returns>
+            /// <exception cref="InvalidOperationException">Thrown if the <paramref name="modelInterface"/> contains no arguments.</exception>
+            private CSharpType TryGetModelArgumentType(CSharpType modelInterface)
```

Comment:
Usually Try* methods will return a bool and have and output parameter with the result that will be set only if the bool is true. In this case, we are throwing if we can't get the argument so I think we can just rename to GetModelArgumentType.

Reply:
fixed.